packages/headless/src/user.ts

```diff
@@ -49,11 +47,14 @@ export function useRegisterUser({
   });
 }
 
+/**
+ * Update all user data in one mutation.
+ */
 export function useUpdateUser({
   onSuccess,
   onError,
 }: {
-  onSuccess?: OnSuccess;
+  onSuccess?: OnSuccess<[IUser.Self, void | null]>;
```

Comment:
No need to return the result of the promise + it is not that clean. 
```suggestion
  onSuccess?: OnSuccess<void>;
```